### DISCUSSION TITLE
SetIf没有起作用

### DISCUSSION THREAD

**Discussion by Melody-TNT** [dotnetcore/FreeSql#1484] [Q&A]
在满足某种条件下，更新某个字段，SetIf报错

**Comment by Melody-TNT** [comment]
![image-20230407092904769](https://user-images.githubusercontent.com/70905243/230523109-82a33cbd-939b-4277-a1a1-7af89697e0ab.png)

**Reply by pjy612** [reply]
是 SetIf 直接就报错了，还是 Execute 的时候报错了？
可以尝试用ToSql 导出 执行的 SQL
如果是 空对象的话 推荐 用 obj?.value

**Reply by Melody-TNT** [reply]
是SetIf直接报错的，原因就是因为meterReading为空，但是我用SetIf的原因就是要不为空的时候更新，如果用直接用obj?.value可能将原来值更新为0

**Reply by pjy612** [reply]
所以说 先用 toSql 看 生成的 sql...
如果功能正常的话 理论上 就不会包含这个字段吧...

**Reply by luoyunchong** [reply]
这是语法问题，SetIf是一个方法，方法传递必然是传的你点的值，当为null时当然会空指针异常。
